Added non-changeable options to rke2-cluster-configuration (#1053)
rancher/rancher-docs: docs/reference-guides/cluster-configuration/rancher-server-configuration/rke2-cluster-configuration.md

        - key2=value2
```

> There are some configuration options which aren't supported to be changed when provisioning via Rancher:

Reviewer: ```suggestion
There are some configuration options that can't be changed when provisioning via Rancher:
```

> - data-dir (Folder to hold state), will always stay the default `/var/lib/rancher/rke2`

Reviewer: ```suggestion
- data-dir (folder to hold state), which defaults to `/var/lib/rancher/rke2`.
```
